src/testing/taylor_green_vortex_energy_check.cpp

```diff
@@ -0,0 +1,40 @@
+#include "taylor_green_vortex_energy_check.h"
+#include "periodic_cube_flow.h"
+
+namespace PHiLiP {
+namespace Tests {
+
+template <int dim, int nstate>
+TaylorGreenVortexEnergyCheckEnergyCheck<dim, nstate>::TaylorGreenVortex(const PHiLiP::Parameters::AllParameters *const parameters_input)
+        : TestsBase::TestsBase(parameters_input)
+{}
+
+template <int dim, int nstate>
+int TaylorGreenVortexEnergyCheck<dim, nstate>::run_test() const
+{
+    std::unique_ptr<FlowSolver<dim,nstate>> flow_solver = FlowSolverFactory<dim,nstate>::create_FlowSolver(parameters_input);
+    std::unique_ptr<FlowSolverCaseBase<dim, nstate>> periodic_cube_flow = std::make_shared<PeriodicCubeFlow<dim,nstate>>(parameters_input);
+
+    // Integrate to final time
+    flow_solver->run_test();
+
+    // Compute kinetic energy
+    const double kinetic_energy_computed = periodic_cube_flow->integrate_over_domain(*(flow_solver->dg),"kinetic_energy");
+    
+    // Hard-coded expected energy value
+    const double kinetic_energy_expected = 1.2074066465958100e-01;
```

Comment:
Can this expected energy value be evaluated based on final time? Or is this just based off some literature? If so, please cite the source.

Reply:
it's a value generated by the code when I run the viscous TGV case, the test is simply to check that any future additions to the code do not modify the "expected" values.